packages/sprotty/src/features/routing/abstract-edge-router.ts

```diff
@@ -297,6 +297,25 @@ export abstract class AbstractEdgeRouter implements IEdgeRouter {
         if (hasChanged) {
             // reset attached elements in index
             edge.index.remove(edge);
+            if (edge.id === edgeInProgressID) {
+                // create a proper edge id after connecting the edge in progress
+                const idGen = (counter: number) => `${edge.sourceId}_to_${edge.targetId}_${counter}`;
+                let idx = 0;
+                let newId = idGen(idx);
+                while (edge.index.getById(newId) !== undefined) {
+                    newId = idGen(++idx);
+                }
+                edge.id = newId;
+                const progressTargetHandle = edge.children.find(child => child.id === edgeInProgressTargetHandleID);
```

Comment:
@spoenemann  I've added a clean-up code for the target handle and the dangling anchor